plugins/MakerbotWriter/MakerbotWriter.py

```diff
@@ -0,0 +1,270 @@
+            if bounds is None:
+                bounds = node_bounds
+            else:
+                bounds = bounds + node_bounds
+
+        if bounds is not None:
+            meta["bounding_box"] = {
+                "x_min": bounds.left,
+                "x_max": bounds.right,
+                "y_min": bounds.back,
+                "y_max": bounds.front,
+                "z_min": bounds.bottom,
+                "z_max": bounds.top,
+            }
+
+        material_bed_temperature = global_stack.getProperty("material_bed_temperature", "value")
+        meta["platform_temperature"] = material_bed_temperature
+
+        build_volume_temperature = global_stack.getProperty("build_volume_temperature", "value")
+        meta["build_plane_temperature"] = build_volume_temperature
+
+        print_information = application.getPrintInformation()
+
+        meta["commanded_duration_s"] = int(print_information.currentPrintTime)
+        meta["duration_s"] = int(print_information.currentPrintTime)
+
+        material_lengths = list(map(meter_to_millimeter, print_information.materialLengths))
+        meta["extrusion_distance_mm"] = material_lengths[0]
+        meta["extrusion_distances_mm"] = material_lengths
+
+        meta["extrusion_mass_g"] = print_information.materialWeights[0]
+        meta["extrusion_masses_g"] = print_information.materialWeights
+
+        meta["uuid"] = print_information.slice_uuid
+
+        materials = []
+        for extruder in extruders:
+            guid = extruder.material.getMetaData().get("GUID")
+            material_name = extruder.material.getMetaData().get("material")
+            material = self._MATERIAL_MAP.get(guid, material_name)
+            materials.append(material)
+
+        meta["material"] = materials[0]
+        meta["materials"] = materials
+
+        materials_temps = [extruder.getProperty("default_material_print_temperature", "value") for extruder in
+                           extruders]
+        meta["extruder_temperature"] = materials_temps[0]
+        meta["extruder_temperatures"] = materials_temps
+
+        meta["model_counts"] = [{"count": 1, "name": node.getName()} for node in nodes]
+
+        tool_types = [MakerbotWriter._EXTRUDER_NAME_MAP.get((name := extruder.variant.getName()), name) for extruder in
+                      extruders]
+        meta["tool_type"] = tool_types[0]
+        meta["tool_types"] = tool_types
+
+        meta["version"] = MakerbotWriter._META_VERSION
+
+        meta["preferences"] = dict()
+        for node in nodes:
+            bounds = node.getBoundingBox()
+            meta["preferences"][str(node.getName())] = {
+                "machineBounds": [bounds.right, bounds.back, bounds.left, bounds.front] if bounds is not None else None,
+                "printMode": CuraApplication.getInstance().getIntentManager().currentIntentCategory,
+            }
+
+        meta["miracle_config"] = {"gaggles": {str(node.getName()): {} for node in nodes}}
```

Comment:
Not sure if leftover debug code, or indicative of the quality of the Makerbot software stack...

Reply:
yes

Reply:
We're working on a MVP at the moment,  it might get used, in the future.  But it needs to be there atm